sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -775,26 +832,31 @@ public void validate(PipelineOptions options) {
           connectionConfiguration != null,
           "ElasticsearchIO.write() requires a connection configuration"
               + " to be set via withConnectionConfiguration(configuration)");
-      checkVersion(connectionConfiguration);
     }
 
     @Override
     public PDone expand(PCollection<String> input) {
-      input.apply(ParDo.of(new WriteFn(this)));
+      int backendVersion = getBackendVersion(getConnectionConfiguration());
```

Comment:
Yes, see my question above about where is the right place to put it in the `WriteFn`

Reply:
It can't be in the constructor, because you're constructing the DoFn at expansion time (`new WriteFn(this)`) - but `@Setup` is a reasonable place to do it.

Reply:
moved the call to `writeFn.setup()`